syllabus/curricula/plr/task_sampler.py

```diff
@@ -363,44 +401,32 @@ def sample(self, strategy=None):
                 if np.random.rand() > self.nu or not proportion_seen < 1.0:
                     return self._sample_replay_level()
 
-            # Otherwise, evaluate a new level
+            # Otherwise, sample a new level
             if self.robust_plr:
                 self.update_with_episode_data(self._evaluate_unseen_level())
                 return self.sample(strategy=strategy)
             else:
-                # Otherwise, sample a new level
                 return self._sample_unseen_level()
 
-
         elif self.replay_schedule == "proportionate":
             if proportion_seen >= self.rho and np.random.rand() < proportion_seen:
                 return self._sample_replay_level()
             else:
                 if self.robust_plr:
-                    while True:
-                        task = self._evaluate_unseen_level()
-                        episode_data = self.evaluate_task(task, self.eval_envs, self.action_value_fn, self.gamma,
-                                                          self.gae_lambda)
-                        self.update_with_episode_data(episode_data, self._average_gae)  # Update task scores
-
-                        # Check if we need to sample another unseen level
-                        num_unseen = (self.unseen_task_weights > 0).sum()
-                        proportion_seen = (self.num_tasks - num_unseen) / self.num_tasks
-                        if proportion_seen < self.rho or np.random.rand() >= proportion_seen:
-                            break
-
-                    return task
+                    self.update_with_episode_data(self._evaluate_unseen_level())
```

Comment:
You call `self.update_with_episode_data` here and within `self._evaluate_unseen_level`. You should only keep one of them.
```suggestion
                    self._evaluate_unseen_level()
```